src/components/layout/ClientForm.tsx

```diff
@@ -0,0 +1,215 @@
+    }, [createToast]);
+
+    const deleteUser = useCallback((user?: User) => {
+        if (user !== undefined && user.id !== undefined) {
+            userService.deleteUser(user.id)
+                .then(r => {
+                    createToast('success', `Compte de ${user.firstName} ${user.lastName} supprimé`);
+                    toggleModal();
+                })
+                .catch(e => createToast('error', e.response ? e.response.data.message : ""));
+        }
+    }, [createToast, toggleModal])
+
+    const openCreditModal = useCallback((user: User) => {
+        if (user.id) {
+            const userSearchEntry: UserSearchEntry = {
+                id: user.id,
+                firstName: user.firstName,
+                lastName: user.lastName,
+                nickName: user.nickName,
+                credit: user.credit
+            }
+            dispatch(initOrder(`${user.firstName} ${user.lastName}`, userSearchEntry));
+            toggleCreditModal();
+        }
+    }, [dispatch, toggleCreditModal])
+
+    const closeCreditModal = useCallback(() => {
+            dispatch(resetOrder());
+            props.refreshUsers();
+            toggleCreditModal();
+    }, [dispatch, toggleCreditModal, props])
+
+    const formik = useFormik({
+        initialValues: {
+            firstName: props.selectedUser === undefined ? '' : props.selectedUser.firstName,
+            lastName: props.selectedUser === undefined ? '' : props.selectedUser.lastName,
+            nickName: props.selectedUser === undefined ? '' : props.selectedUser.nickName,
+            credit: props.selectedUser === undefined ? undefined : props.selectedUser.credit,
+            section: props.selectedUser === undefined ? Section.FIRST : props.selectedUser.section,
+            isMembership: true
+        },
+        onSubmit: values => {
+            const userToSave: User = {
+                id: props.selectedUser ? props.selectedUser.id : undefined,
+                firstName: values.firstName,
+                lastName: values.lastName,
+                nickName: values.nickName,
+                credit: values.credit ? values.credit : 0,
+                section: values.section,
+                isMembership: true
+            }
+
+            const save = userToSave.id ? updateUser(userToSave) : addUser(userToSave);
+
+            save.then(() => {
+                props.refreshUsers()
+                formik.resetForm();
+            });
+        },
+        enableReinitialize: true
+    });
+
+    return (
+        <div>
+            <h3>{props.selectedUser ? "Mettre à jour un client" : "Créer un client"}</h3>
+            <div className="user-form">
+                <form onSubmit={formik.handleSubmit}>
+                    <div className="user-form-line">
+                        <input
+                            className="user-form-input"
+                            id="firstName"
```

Comment:
Les ids ne sont pas nécessaires si tu n'a pas de label

```suggestion
<label htmlFor="firstName">Firstname</label>
<input id="firstname" .../>
```

Reply:
Je trouve sans label c'est plus lisible dans l'interface